adapters/openrtb_util_test.go

```diff
@@ -26,26 +49,229 @@ func TestOpenRTB(t *testing.T) {
+				},
+			},
+		},
+	}
+	resp, err := makeOpenRTBGeneric(&pbReq, &pbBidder, "test", []pbs.MediaType{pbs.MEDIA_TYPE_VIDEO, pbs.MEDIA_TYPE_BANNER}, false)
+	assert.Equal(t, err, nil)
+	assert.Equal(t, len(resp.Imp), 1)
+	assert.Equal(t, resp.Imp[0].ID, "unitCode")
+	assert.EqualValues(t, resp.Imp[0].Banner.W, 10)
+	assert.EqualValues(t, resp.Imp[0].Video.W, 10)
+	assert.EqualValues(t, resp.Imp[0].Video.MaxDuration, 30)
+	assert.EqualValues(t, resp.Imp[0].Video.MinDuration, 15)
+}
+
+func TestOpenRTBMultiMediaImpFiltered(t *testing.T) {
+
+	pbReq := pbs.PBSRequest{}
+	pbBidder := pbs.PBSBidder{
+		BidderCode: "bannerCode",
+		AdUnits: []pbs.PBSAdUnit{
+			{
+				Code:       "unitCode",
+				MediaTypes: []pbs.MediaType{pbs.MEDIA_TYPE_VIDEO, pbs.MEDIA_TYPE_BANNER},
+				Sizes: []openrtb.Format{
+					{
+						W: 10,
+						H: 12,
+					},
+				},
+				Video: pbs.PBSVideo{
+					Mimes:          []string{"video/mp4"},
+					Minduration:    15,
+					Maxduration:    30,
+					Startdelay:     5,
+					Skippable:      0,
+					PlaybackMethod: 1,
+				},
+			},
+		},
+	}
+	resp, err := makeOpenRTBGeneric(&pbReq, &pbBidder, "test", []pbs.MediaType{pbs.MEDIA_TYPE_BANNER}, false)
+	assert.Equal(t, err, nil)
+	assert.Equal(t, len(resp.Imp), 1)
+	assert.Equal(t, resp.Imp[0].ID, "unitCode")
+	assert.EqualValues(t, resp.Imp[0].Banner.W, 10)
+	assert.EqualValues(t, resp.Imp[0].Video, (*openrtb.Video)(nil))
+}
+
+func TestOpenRTBSingleMediaImp(t *testing.T) {
+
+	pbReq := pbs.PBSRequest{}
+	pbBidder := pbs.PBSBidder{
+		BidderCode: "bannerCode",
+		AdUnits: []pbs.PBSAdUnit{
+			{
+				Code:       "unitCode",
+				MediaTypes: []pbs.MediaType{pbs.MEDIA_TYPE_VIDEO, pbs.MEDIA_TYPE_BANNER},
+				Sizes: []openrtb.Format{
+					{
+						W: 10,
+						H: 12,
+					},
+				},
+				Video: pbs.PBSVideo{
+					Mimes:          []string{"video/mp4"},
+					Minduration:    15,
+					Maxduration:    30,
+					Startdelay:     5,
+					Skippable:      0,
+					PlaybackMethod: 1,
+				},
+			},
+		},
+	}
+	resp, err := makeOpenRTBGeneric(&pbReq, &pbBidder, "test", []pbs.MediaType{pbs.MEDIA_TYPE_VIDEO, pbs.MEDIA_TYPE_BANNER}, true)
+	assert.Equal(t, err, nil)
+	assert.Equal(t, len(resp.Imp), 2)
+	assert.Equal(t, resp.Imp[0].ID, "unitCode")
+	assert.EqualValues(t, resp.Imp[0].Video.MaxDuration, 30)
+	assert.EqualValues(t, resp.Imp[0].Video.MinDuration, 15)
+	assert.Equal(t, resp.Imp[1].ID, "unitCode")
+	assert.EqualValues(t, resp.Imp[1].Banner.W, 10)
+}
+
 func TestOpenRTBNoSize(t *testing.T) {
 
 	pbReq := pbs.PBSRequest{}
 	pbBidder := pbs.PBSBidder{
 		BidderCode: "bannerCode",
 		AdUnits: []pbs.PBSAdUnit{
 			{
-				Code: "unitCode",
+				Code:       "unitCode",
+				MediaTypes: []pbs.MediaType{pbs.MEDIA_TYPE_BANNER},
 			},
 		},
 	}
-	resp := makeOpenRTBGeneric(&pbReq, &pbBidder, "test")
-	assert.Equal(t, resp.Imp[0].ID, "")
+	resp, err := makeOpenRTBGeneric(&pbReq, &pbBidder, "test", []pbs.MediaType{pbs.MEDIA_TYPE_BANNER}, true)
+	//	assert.Equal(t, resp.Imp[0].ID, "")
```

Comment:
can delete the dead code or uncomment it

Reply:
Removed